lib/puppet/resource_api/parameter.rb

```diff
@@ -0,0 +1,50 @@
+require 'puppet/util'
+require 'puppet/parameter'
+
+module Puppet; module ResourceApi; end; end # predeclare the main module # rubocop:disable Style/Documentation,Style/ClassAndModuleChildren
+
+# Class containing parameter functionality for ResourceApi.
+class Puppet::ResourceApi::Parameter < Puppet::Parameter
+  # This initialize takes arguments and sets up new parameter.
+  # @param type_name the name of the Puppet Type
+  # @param data_type the data type of parameter instance
+  # @param attribute_name the name of attribue of the parameter
+  # @param resource_hash the resource hash instance which is passed to the
+  # parent class.
+  def initialize(type_name, data_type, attribute_name, resource_hash)
+    @type_name = type_name
+    @data_type = data_type
+    @attribute_name = attribute_name
+    super(resource_hash) # Pass resource to parent Puppet class.
+  end
+
+  # This method handles return of assigned value from parameter.
+  # @return [value] the type value
+  def value # rubocop:disable Style/TrivialAccessors
```

Comment:
Missed that on the first round: the entire method can be replaced by 
```ruby
attr_reader :value
```

By convention as the first thing in the class.